homework-submission/week1/src/App.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import Todos from './components/Todos';
+import AddTodo from './components/AddTodo';
+import data from './sources/data.json'
+
+class App extends Component {
+  state = {
+    data
+  }
+    handler = e => {
+      let todos = this.state.data;
+      todos[e.target.id -1].done = !todos[e.target.id -1].done;
+      this.setState({
+        todos
+      })
+    }
+
+    addTodo = e => {
+      e.preventDefault();
+      const {data} = this.state;
+      const newTodo = {
+        id: data.length + 1,
+        description: e.target.description.value,
+        deadline: e.target.deadline.value,
+        done: false
+      };
+      data.push(newTodo)
+      this.setState({
+        data
+      })
+      e.target.description.value = "";
+      e.target.deadline.value = "";
+    }
+
+    removeTodo = (i) => {
+      this.state.data.splice(i, 1);
```

Comment:
The same here, you are deleting the todo from the original state!. Please fix this.